### PR TITLE
Add test feature: generate a report from an agent

### PR DESCRIPTION
**Pull request by Mayons95**
### Description
Hi team, this PR includes a new test case to validate the report generation from any module choosing a particular agent.
This validation will increase the testing coverage on all modules.
 
### Issues Resolved
https://github.com/wazuh/wazuh-kibana-app/issues/4627

### Evidence
Test it's working for WZD and Kibana

image

### Test comments:
The errors during the test execution are not related to this PR. 
It's a bug previously detected and reported on: https://github.com/wazuh/wazuh-kibana-app/issues/4621

### CODE REVIEW

File: test/cypress/cypress/integration/step-definitions/reporting/the-user-agent-report-is-displayed-in-the-table-when.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Then } from 'cypress-cucumber-preprocessor/steps';
```
Comment (mauceballos) [inline test/cypress/cypress/integration/step-definitions/reporting/the-user-agent-report-is-displayed-in-the-table-when.js:1]: the name of the file "the-user-agent-report-is-displayed-in-the-table-when.js" 
should be 
 "the-user-agent-report-is-displayed-in-the-table-then.js"

**Review by mauceballos — COMMENTED** [review]
please, see the comments thx.

File: test/cypress/cypress/integration/step-definitions/reporting/the-user-agent-report-is-displayed-in-the-table-when.js (hunk 1 above)
Comment (Mayons95) [reply test/cypress/cypress/integration/step-definitions/reporting/the-user-agent-report-is-displayed-in-the-table-when.js:1]: I'll take it

Comment (Mayons95) [reply test/cypress/cypress/integration/step-definitions/reporting/the-user-agent-report-is-displayed-in-the-table-when.js]: Done !

**Review by mauceballos — APPROVED** [review]

**Review by Machi3mfl — APPROVED** [review]

**Pull request merged**